docs: add information about obtaining VRN (#130)
vtex-apps/recommendation-shelf: docs/README.md

### Configuration

Configure the `recommendation-shelf` block in your theme app using the following props:

### Props

Configure the `recommendation-shelf` block using the following properties:

| Prop name | Type | Description | Default value |
| - | - | - | - |
| `title` | `string` | Shelf title displayed to users. | - |
| `campaignVrn` | `string` | VRN for the recommendation campaign (the recommendation **list ID** from Admin. See [Obtaining the VRN](#obtaining-the-vrn)). | - |
| `displayTitle` | `boolean` | Whether to show the shelf title (`true`) or hide it (`false`). | `true` |
| `itemsContext` | `array` | Context source for items in the recommendation request (`PDP` or `CART`). Useful for enabling shelves on the cart page with `CROSS_SELL`. | `['PDP']` |

### Obtaining the VRN

> The `campaignVrn` prop expects the **recommendation list ID** from VTEX Admin, the same value you get when you copy the list identifier after creating a recommendation list (that string is your VRN for the shelf).

Reviewer: ```suggestion
The campaignVrn prop takes the recommendation list ID from VTEX Admin. This is the identifier string shown after you create a recommendation list, also referred to as the shelf's VRN.
```

> #### Copying the list ID

Reviewer: I found this section confusing. It sort of repeats the paragraph above and the exact order of the actions needed is not very precise. I would recommend something like:

> Once you create a list using the steps described in this tutorial [...], you can copy [...].
> [...]
> If you need to copy the list ID of a list that already exists, follow these steps.

Reviewer: I am not sure my example was super helpful, but anyway, I think there's room for improvement.

After creating the list, copy the ID using either path below:

- On the confirmation screen **List created successfully**, click **Copy ID**. The ID is copied to your clipboard.
- Or copy it later from the [Recommendations dashboard](https://help.vtex.com/en/docs/tutorials/recommendations-dashboard):

  1. Go to **Storefront > Recommendations**.
  2. Find the desired list in the shelf table.
  3. Click the ⋮ menu on the shelf row.
  4. Select **Copy ID**.

See also:

* [Creating recommendation lists](https://help.vtex.com/en/docs/tutorials/creating-recommendation-lists)
* [Product Recommendations](https://help.vtex.com/docs/tutorials/product-recommendations-beta)

## Recommendation strategies

The **`campaignVrn`** string must match `vrn:recommendations:<account>:<campaign-type>:<campaign-id>`. The **`campaign-type`** segment maps to an internal **`RecommendationType`** used when calling recommendations:

| VRN `campaign-type` | Resolved `RecommendationType` | Description |
| ------------------- | ------------------------------ | ----------- |

> | `rec-cross-v1`, `rec-cross-v2` | `CROSS_SELL` | Complementary products (often bought together); use `itemsContext` `CART` on the cart page when needed. |

Reviewer: ```suggestion
| `rec-cross-v1`, `rec-cross-v2` | `CROSS_SELL` | Complementary products (often bought together). Use `itemsContext` `CART` on the cart page when needed. |
```

| `rec-similar-v1`, `rec-similar-v2` | `SIMILAR_ITEMS` | Similar-item recommendations for the current product context. |
| `rec-persona-v1`, `rec-persona-v2` | `PERSONALIZED` | Personalized recommendations from shopper behavior. |
| `rec-last-v1`, `rec-last-v2` | `LAST_SEEN` | Recently viewed products for the shopper. |
| `rec-top-items-v1`, `rec-top-items-v2` | `TOP_ITEMS` | Popular / top-performing products in the store. |
| `rec-search-v2` | `SEARCH_BASED` | Search-driven recommendations. |

> `v1` and `v2` denote different generations of the same strategy kind; use the **`campaignVrn`** copied from Admin for your list.

Reviewer: ```suggestion
`v1` and `v2` denote different generations of the same strategy kind. Use the **`campaignVrn`** copied from Admin for your list.
```
